src/erlcloud_ec2.erl

```diff
@@ -3574,31 +3583,65 @@ ec2_query(Config, Action, Params, ApiVersion) ->
                                   Config#aws_config.ec2_port,
                                   "/", QParams, "ec2", Config).
 
-default_config() -> erlcloud_aws:default_config().
+% Exported Query Function with parameter handling
+% Query takes in: 
+% - an aws_config
+%   - an ec2 action string: list of possible are found here https://docs.aws.amazon.com/AWSEC2/latest/APIReference/OperationList-query-ec2.html
+% - a map of query parameters: An example can be found here https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeImages.html
+    % NOTE: Any fields in the documentation that end in ".N", may just store a list - No need to include ".N".   
+
+-spec query(aws_config(), string(), map(), query_opts()) -> ok_error().
+query(Config, Action, Params, Opts) ->
+    % NewOpts = maps:merge(?OPT_DEFAULTS, Params),
+    ApiVersion= maps:get(version, Opts, ?NEW_API_VERSION),
+    Filter = maps:get(filter, Opts, []),
+    ResponseFormat = maps:get(response_format, Opts, undef),
```

Comment:
Another `undef` vs `none` here. I'd lean toward defining this spec in a common place, maybe erlcoud_aws, and referencing the exported spec.